### PR TITLE
Get eventloop from event_loop_policy to avoid deprecation warning.

### LINKED ISSUE #5685: DeprecationWarning: There is no current event loop
```
/home/runner/work/scrapy/scrapy/scrapy/utils/reactor.py:70: DeprecationWarning: There is no current event loop
  event_loop = asyncio.get_event_loop()
tests/test_downloadermiddleware.py: 2 warnings
tests/test_utils_asyncgen.py: 2 warnings
tests/test_utils_defer.py: 4 warnings
tests/test_utils_python.py: 2 warnings
tests/test_utils_signal.py: 2 warnings
  /home/runner/work/scrapy/scrapy/scrapy/utils/defer.py:272: DeprecationWarning: There is no current event loop
    return Deferred.fromFuture(asyncio.ensure_future(o))
tests/test_utils_asyncio.py::AsyncioTest::test_install_asyncio_reactor
  /home/runner/work/scrapy/scrapy/scrapy/utils/reactor.py:70: DeprecationWarning: There is no current event loop
    event_loop = asyncio.get_event_loop()
```

Not sure yet what does this imply and when can this break (this is from tests on 3.10 and 3.11).

### PR DESCRIPTION
**Pull request by Godson-Gnanaraj**
Fixes #5685

Use `policy.get_event_loop()` instead of `asyncio.get_event_loop()`.  
If there is no event loop running, policy will create new_event_loop.

### CODE REVIEW

File: scrapy/utils/reactor.py [hunk 1]
```diff
@@ -58,16 +58,19 @@ def install_reactor(reactor_path, event_loop_path=None):
     reactor_class = load_object(reactor_path)
     if reactor_class is asyncioreactor.AsyncioSelectorReactor:
         with suppress(error.ReactorAlreadyInstalledError):
+            policy = asyncio.get_event_loop_policy()
             if sys.version_info >= (3, 8) and sys.platform == "win32":
```
Comment (elacuesta) [inline scrapy/utils/reactor.py:62]: This condition could now be refactored as
```python
if (
    sys.version_info >= (3, 8)
    and sys.platform == "win32"
    and not isinstance(policy, asyncio.WindowsSelectorEventLoopPolicy)
):
```

**Review by Gallaecio — COMMENTED** [review]
Could you add a test for this change?

**Comment by Godson-Gnanaraj** [comment]
> Could you add a test for this change?

I think windows tests should cover these changes. But I couldn't see windows coverage reports being uploaded.

**Comment by Gallaecio** [comment]
I meant a test to verify that the warning is no longer triggered, for which I imagine there is no existing test.

Good point about coverage, though. We need to look into that.

**Comment by wRAR** [comment]
```
tests/test_downloadermiddleware.py: 2 warnings
tests/test_utils_asyncgen.py: 2 warnings
tests/test_utils_defer.py: 4 warnings
tests/test_utils_python.py: 2 warnings
tests/test_utils_signal.py: 2 warnings
  /home/runner/work/scrapy/scrapy/scrapy/utils/defer.py:272: DeprecationWarning: There is no current event loop
    return Deferred.fromFuture(asyncio.ensure_future(o))
```

**Comment by wRAR** [comment]
Oh right, you said it's a partial fix.

**Review by wRAR — APPROVED** [review]

**Comment by Godson-Gnanaraj** [comment]
> ```
> tests/test_downloadermiddleware.py: 2 warnings
> tests/test_utils_asyncgen.py: 2 warnings
> tests/test_utils_defer.py: 4 warnings
> tests/test_utils_python.py: 2 warnings
> tests/test_utils_signal.py: 2 warnings
>   /home/runner/work/scrapy/scrapy/scrapy/utils/defer.py:272: DeprecationWarning: There is no current event loop
>     return Deferred.fromFuture(asyncio.ensure_future(o))
> ```

@wRAR  Can I move `get_event_loop` to [scrapy/utils/misc.py](https://github.com/scrapy/scrapy/blob/master/scrapy/utils/misc.py) so that it can be used in both [reactor.py](https://github.com/scrapy/scrapy/blob/master/scrapy/utils/reactor.py) and [defer.py](https://github.com/scrapy/scrapy/blob/master/scrapy/utils/defer.py)?

**Comment by wRAR** [comment]
What do you mean by `get_event_loop_policy`? Some new method?

**Comment by wRAR** [comment]
I think it's fine to put it into utils.reactor? utils.defer already imports utils.reactor.

**Comment by wRAR** [comment]
Should the remaining warning be fixed by passing `loop=policy.get_event_loop())` to `asyncio.ensure_future(o)` in `deferred_from_coro()`?

**Review by wRAR — APPROVED** [review]
Perfect!

**Review by Gallaecio — COMMENTED** [review]
The code looks good to me.

Can someone manually verify that the new test fails if the other changes are reverted? That’s the only thing I wanted to check manually when I found the time.

**Comment by wRAR** [comment]
@Gallaecio I've change the test in the master branch and ran it, I got `AssertionError: 1 != 0`

**Review by Gallaecio — APPROVED** [review]
@Godson-Gnanaraj Thanks!

**Pull request merged**